### PR TITLE
Foreign Contacts - Othernames and birthplace (part 2)

### LINKED ISSUE #2536: Foreign contacts - Other names used
- [x] Foreign national other names used inputs are not being saved
- [x] Foreign national's place of birth - country not being saved

screen shot 2018-01-29 at 12 27 11 pm
screen shot 2018-01-29 at 12 28 27 pm

### PR DESCRIPTION
**Pull request by floresj**
Refactored country handling based on discussion in PR #2549. Just created a new branch to start fresh. Resolves #2536.

- Foreign national other names used inputs are not being saved
- Foreign national's place of birth - country not being saved 
  - This seemed to be site wide. There are instances where we only have a simple string country value. The Location component wasn't handling this case so countries would be blank.

### CODE REVIEW

File: src/components/Form/Country/Country.jsx [hunk 1]
```diff
@@ -11,7 +11,6 @@ export default class Country extends ValidationElement {
     super(props)
 
     this.state = {
-      value: props.simpleValue,
```
Comment (floresj) [inline src/components/Form/Country/Country.jsx]: No longer used.

File: src/components/Form/Location/Address.jsx [hunk 2]
```diff
@@ -380,7 +381,7 @@ export default class Address extends ValidationElement {
                 <Country name="country"
                          className="required"
                          label={this.props.countryLabel}
-                         {...this.props.country}
+                         {...countryValueResolver(this.props)}
```
Comment (floresj) [inline src/components/Form/Location/Address.jsx]: Use `countryValueResolver` to appropriately resolves and returns the parameter that the `<Country />` component is expecting. When the backend returns information, `this.props.country` is a string. However, when updating the application from the client app, `this.props.country` is an object. This method ensures that return object is normalized.

File: src/components/Form/Location/ToggleableLocation.jsx [hunk 3]
```diff
@@ -58,19 +58,22 @@ export default class ToggleableLocation extends ValidationElement {
   }
 
   updateStreet (values) {
-    this.update({ street: values.value})
+    this.update({street: values.value})
   }
 
   updateCity (values) {
-    this.update({ city: values.value})
```
Comment (floresj) [inline src/components/Form/Location/ToggleableLocation.jsx]: evil space.

File: src/components/Section/Foreign/Contacts/ForeignNational.jsx [hunk 4]
```diff
@@ -489,11 +482,13 @@ export default class ForeignNational extends ValidationElement {
                           required={this.props.required}
                           onError={this.props.onError}
                           scrollIntoView={this.props.scrollIntoView}>
-          <Field title={i18n.t('foreign.contacts.heading.aliasname')}
-                 optional={true}
-                 scrollIntoView={this.props.scrollIntoView}>
-            <Name name="Alias" bind={true} required={this.props.required} scrollIntoView={this.props.scrollIntoView} />
-          </Field>
+          <AccordionItem scrollIntoView={this.props.scrollIntoView}>
```
Comment (floresj) [inline src/components/Section/Foreign/Contacts/ForeignNational.jsx]: I tried to keep your editors tab format. Didn't want to redo the entire file lol.

File: api/model/form/location.go [hunk 5]
```diff
@@ -29,16 +29,17 @@ const (
...
+	ID              int    `json:"-"`
+	Layout          string `json:"layout"`
+	Street1         string `json:"street,omitempty"`
+	Street2         string `json:"street2,omitempty"`
+	City            string `json:"city,omitempty"`
+	State           string `json:"state,omitempty"`
+	Zipcode         string `json:"zipcode,omitempty"`
+	County          string `json:"county,omitempty"`
+	Country         string `json:"country,omitempty"`
+	CountryComments string `json:"countryComments,omitempty"`
```
Comment (floresj) [inline api/model/form/location.go]: Added new `countryComments` field to struct.

File: src/components/Section/Foreign/Contacts/ForeignNational.jsx (hunk 4 above)
Comment (bmallred) [reply src/components/Section/Foreign/Contacts/ForeignNational.jsx]: LOL it is all good.

**Pull request merged**